### PR TITLE
feat: initial vue2 support

### PR DESCRIPTION
**Pull request by sibbngheid**
/cc @antfu I need help with types 😢😢

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -65,6 +71,9 @@ export const extractor = (content: string) => {
   return broadMatches.concat(innerMatches, extract);
 };
 
-export default (app: App, directiveName = "variantwind") => {
-  app.directive(directiveName, directive);
+export default (app: App | VueConstructor, directiveName = "variantwind") => {
+  if (app.version[0] === "3") {
+    (app as App).directive(directiveName, directive);
+  }
+  (app as VueConstructor).directive(directiveName, directive2);
```
Comment (sibbngheid) [inline index.ts:78]: I'm curious about is this can be done better?

**Comment by antfu** [comment]
Is there any thing I can help with?

**Comment by sibbngheid** [comment]
The `(app as App)` and `(app as VueConstructor)` part is seems wrong for me. Is it can't be auto infer type some how based on my if condition (`app.version[0] === "3"`) ?

**Comment by antfu** [comment]
Blind coding, hope it's right🤣
```ts
const isVue3 = <T extends App | VueConstructor>(app: T): T is App => app.version[0] === "3"

if (isVue3(app)) { 
  // App
}
else {
  // VueConstructor
}
```

**Comment by sibbngheid** [comment]
image

**Comment by antfu** [comment]
Euh, change it to `app is App`

**Comment by sibbngheid** [comment]
> Euh, change it to `app is App`

Already tried, but another type error 😢
image

**Comment by antfu** [comment]
App<any>

**Comment by sibbngheid** [comment]
> Try `App<any>`?

Didn't help 😔

**Comment by antfu** [comment]
```ts
const isVue3 = (app: App | VueConstructor): app is App => app.version[0] === "3"
```
?

**Comment by sibbngheid** [comment]
> How about
> 
> ```ts
> const isVue3 = (app: App | VueConstructor): app is App => app.version[0] === "3"
> ```

✨This is magic ✨

Works 🤗

**Pull request merged**